# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1205205527 14400
#      Mon Mar 10 23:18:47 2008 -0400
# Node ID ba8fcc115fee525b479d94ac5a53115ecf7ee257
# Parent  7ebdc99a0babb6a33a585337b9e5165801c383ae
mex.cc: arg to mxCreateLogicalScalar is now mxLogical

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,10 +1,12 @@
 2008-03-10  John W. Eaton  <jwe@octave.org>
 
+	* mex.cc (mxCreateLogicalScalar): Argument is now mxLogical.
+
 	* data.cc (Fisfloat): New function.
 	* ov.h (octave_value::is_float_type): Ditto.
 	* ov-base.h (octave_base_value::is_float_type): Ditto.
 	* ov-complex.h (octave_complex): Ditto.
 	* ov-cx-mat.h (octave_complex_matrix): Ditto.
 	* ov-cx-sparse.h (octave_sparse_complex_matrix): Ditto.
 	* ov-range.h (octave_range): Ditto.
 	* ov-re-mat.h (octave_matrix): Ditto.
diff --git a/src/mex.cc b/src/mex.cc
--- a/src/mex.cc
+++ b/src/mex.cc
@@ -2539,17 +2539,17 @@ mxCreateLogicalArray (mwSize ndims, cons
 
 mxArray *
 mxCreateLogicalMatrix (mwSize m, mwSize n)
 {
   return maybe_mark_array (new mxArray (mxLOGICAL_CLASS, m, n));
 }
 
 mxArray *
-mxCreateLogicalScalar (int val)
+mxCreateLogicalScalar (mxLogical val)
 {
   return maybe_mark_array (new mxArray (mxLOGICAL_CLASS, val));
 }
 
 mxArray *
 mxCreateNumericArray (mwSize ndims, const mwSize *dims, mxClassID class_id,
 		      mxComplexity flag)
 {
